src/device-state/current-state.ts

```diff
@@ -2,55 +2,47 @@ import * as _ from 'lodash';
 import * as url from 'url';
 import { CoreOptions } from 'request';
 
-import * as constants from '../lib/constants';
+import { backoffPromise } from '../lib/backoff';
 import { log } from '../lib/supervisor-console';
 import { InternalInconsistencyError, StatusError } from '../lib/errors';
 import { getRequestInstance } from '../lib/request';
 import * as sysInfo from '../lib/system-info';
+import sleep from '../lib/sleep';
 
 import { DeviceStatus } from '../types/state';
 import * as config from '../config';
-import { SchemaTypeKey, SchemaReturn } from '../config/schema-type';
 import * as eventTracker from '../event-tracker';
 import * as deviceState from '../device-state';
 
-// The exponential backoff starts at 15s
-const MINIMUM_BACKOFF_DELAY = 15000;
-
 const INTERNAL_STATE_KEYS = [
 	'update_pending',
 	'update_downloaded',
 	'update_failed',
 ];
 
 export let stateReportErrors = 0;
+
 const lastReportedState: DeviceStatus = {
 	local: {},
 	dependent: {},
 };
-const stateForReport: DeviceStatus = {
-	local: {},
-	dependent: {},
-};
-let reportPending = false;
 
-type CurrentStateReportConf = {
-	[key in keyof Pick<
-		config.ConfigMap<SchemaTypeKey>,
-		| 'uuid'
-		| 'apiEndpoint'
-		| 'apiTimeout'
-		| 'deviceApiKey'
-		| 'deviceId'
-		| 'localMode'
-		| 'hardwareMetrics'
-	>]: SchemaReturn<key>;
-};
+/**
+ * patchInterval is set when startReporting successfuly queries the config
+ */
+let patchInterval: number;
 
-type StateReport = {
```

Comment:
Keeping a local copy of state for report didn't make sense because it would become outdated theoretically right away if something changes. The other types didn't seem necessary either.